src/Avalonia.Controls/Primitives/TemplatedControl.cs

```diff
@@ -290,12 +290,6 @@ public sealed override void ApplyTemplate()
                     ApplyTemplatedParent(child, this);
                     ((ISetLogicalParent)child).SetParent(this);
                     VisualChildren.Add(child);
-                    
-                    // Existing code kinda expect to see a NameScope even if it's empty
-                    if (nameScope == null)
-                    {
-                        nameScope = new NameScope();
-                    }
 
```

Comment:
I am unsure if namescope won't be ever null here.
Custom template technically could return null as a namescope. Unless we rely on nullability attributes in the new user's code, and assume that it's rare to have custom templates.

Reply:
`ITemplateResult.NameScope` was already correctly annotated as non-null. Everyone is still free to ignore any annotation, but shouldn't complain too much when things break imo :)